clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11755,9 +11758,12 @@ def err_sycl_function_attribute_mismatch : Error<
   "SYCL kernel without %0 attribute can't call a function with this attribute">;
 def err_sycl_x_y_z_arguments_must_be_one : Error<
   "all %0 attribute arguments must be '1' when the %1 attribute argument is '0'">;
-def err_sycl_attribute_internal_function
+def err_sycl_attribute_internal_decl
     : Error<"%0 attribute cannot be applied to a "
-            "static function or function in an anonymous namespace">;
+            "static %select{function|variable}1 or %select{function|variable}1 "
+            "in an anonymous namespace">;
+def err_sycl_attribute_not_device_global
+    : Error<"%0 attribute can only appear on 'device_global' variables">;
```

Comment:
`appear on` sounds odd. I think `can only be applied to` sounds better.

Reply:
Sounds good to me. Thanks.

Reply:
Is there a test for this diagnostic on a function?

Reply:
@premanandrao , could you please elaborate a bit what exactly do you mean? `SYCL_EXTERNAL` macro can be legally applied to functions.

Reply:
Sorry, looks like my comment got applied at the wrong place.
I meant for the previous diagnostic: ``err_sycl_external_global`` - if there were test cases where the '%1' argument is a function.   And to answer my own question, looks like they already exist and your changes don't affect it, just generlaizes them. 

So all set.  :-)